Fixes #20365: Fix schema and field definitions for OpenAPI

pheus opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #20365

- Implement `get_internal_type()` on custom fields for Django introspection.
- Annotate `id`/`name` path params and add explicit `operation_id`s for queues/workers/tasks
- Provide `get_serializer_class()` and `get_serializer_context()` on the RQ base viewset

jeremystretch commented on netbox/core/api/views.py:
```diff
@@ -95,7 +95,7 @@ class ObjectTypeViewSet(ReadOnlyModelViewSet):
 
 
-class BaseRQViewSet(viewsets.ViewSet):
+class BaseRQViewSet(viewsets.GenericViewSet):
```
I'm not sure it makes sense to employ GenericViewSet here as it adds methods such as `get_queryset()` which aren't applicable, as the view doesn't use a model. Could we instead just add declare a `get_serializer_class()` method?

jeremystretch commented on netbox/core/api/views.py:
```diff
@@ -199,7 +215,11 @@ def get_task_from_id(self, task_id):
-    @extend_schema(responses={200: OpenApiTypes.OBJECT})
+    @extend_schema(
+        operation_id='core_background_tasks_retrieve_by_id',
+        parameters=[OpenApiParameter(name='pk', type=OpenApiTypes.STR, location=OpenApiParameter.PATH)],
```
Is this redundant to line 192 above? (Also, here we call the parameter `pk` but above it's named `id`.)

pheus committed:
fix(api): Fix schema and field definitions for OpenAPI

Add `get_internal_type()` to custom field classes for Django compatibility,
annotate path parameters and operation IDs for background endpoints, and
provide serializer context on the RQ base viewset to clear schema warnings.

Fixes #20365

jeremystretch reviewed (approved):
Thanks @pheus!
